hack/lib/golang.sh

```diff
@@ -538,6 +538,7 @@ EOF
   local go_version
   IFS=" " read -ra go_version <<< "$(GOFLAGS='' go version)"
   local minimum_go_version
+  # TODO: Update to go1.23 as soon we are ready to merge this
   minimum_go_version=go1.22
```

Comment:
@cpanato are we good to change this?

Reply:
i think we change that when the 1.23 is official released not in rcs